source/common/common/logger.h

```diff
@@ -396,16 +396,26 @@ template <Id id> class Loggable {
 
 namespace Utility {
 
+constexpr static absl::string_view TagsPrefix = "[Tags: ";
+constexpr static absl::string_view TagsSuffix = "] ";
+constexpr static absl::string_view TagsSuffixForSearch = "\"] ";
+
 /**
  * Sets the log format for a specific logger.
  */
 void setLogFormatForLogger(spdlog::logger& logger, const std::string& log_format);
 
 /**
  * Serializes custom log tags to a string that will be prepended to the log message.
+ * To support JSON format tags, the key and values will be serialized as escaped JSON string.
```

Comment:
Seems like non-JSON users are potentially paying a large cost for this? Can this be done only for the JSON case somehow?

Reply:
thanks, done